src/main/java/org/crac/Core.java

```diff
@@ -182,31 +182,29 @@ public void register(Resource resource) {
         }
     }
 
-    static class CompatMaster extends Compat {
-        CompatMaster() throws ClassNotFoundException, NoSuchMethodException, IllegalAccessException,
-                InvocationTargetException {
-            super("javax.crac");
+    static Compat loadCompat(String packageName) {
+        try {
+            return new Compat(packageName);
+        } catch (Throwable t) {
+            return null;
         }
     }
 
-    static class CompatJdk extends Compat {
-        CompatJdk() throws ClassNotFoundException, NoSuchMethodException, IllegalAccessException,
-                InvocationTargetException {
-            super("jdk.crac");
+    static {
+        Compat candidate = null;
+        String propCompatImpl = System.getProperty("org.crac.Core.Compat");
```

Comment:
would it be valuable to completely disable crac core and don't load anything? or don't allow to use default if required implementation is not found?